lessons/ro/chapter_7.yaml

```
@@ -0,0 +1,271 @@
+
+    Un obiect al trăsăturii este o structura ce conține un pointer al instanței noastre, alături de o lista
+    de pointeri la funcții către metodele acesteia.
+
+    Detalii:
+
+    * Această lista de funcții este cunoscută în C++ sub denumirea de *vtable*.
+- title: Manevrarea Datelor Fără Dimensiune
+  content_markdown: >
+    Trăsăturile introduc o provocare interesantă când dorim să le stocăm în interiorul altei structuri.
+
+    Trăsăturile ascund structura originală, astfel ascunzând și dimensiunea originală. Valorile fără dimensiune
+    sunt manevrate în două moduri în Rust:
+
+    * tipuri generice - Folosind tipuri parametrizate creăm structuri/funcții cu tipuri și dimensiuni cunoscute
+
+    * indirectie - Punerea instanțelor pe heap ne oferă un nivel de indirectie ce ne permite a nu ne îngrijora
+    cu privire la dimensiunea tipului real, ci doar să stocăm un pointer către acesta.
+    
+    Mai există și alte modalități!
+- title: Funcții Generice
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=struct%20CreaturaMaritima%20%7B%0A%20%20%20%20pub%20nume%3A%20String%2C%0A%20%20%20%20zgomot%3A%20String%2C%0A%7D%0A%0Aimpl%20CreaturaMaritima%20%7B%0A%20%20%20%20pub%20fn%20luare_zgomot(%26self)%20-%3E%20%26str%20%7B%0A%20%20%20%20%20%20%20%20%26self.zgomot%0A%20%20%20%20%7D%0A%7D%0A%0Atrait%20FacatorDeZgomot%20%7B%0A%20%20%20%20fn%20fa_zgomot(%26self)%3B%0A%7D%0A%0Aimpl%20FacatorDeZgomot%20for%20CreaturaMaritima%20%7B%0A%20%20%20%20fn%20fa_zgomot(%26self)%20%7B%0A%20%20%20%20%20%20%20%20println!(%22%7B%7D%22%2C%20%26self.luare_zgomot())%3B%0A%20%20%20%20%7D%0A%7D%0A%0Afn%20generic_fa_zgomot%3CT%3E(creatura%3A%20%26T)%0Awhere%0A%20%20%20%20T%3A%20FacatorDeZgomot%2C%0A%7B%0A%20%20%20%20%2F%2F%20cunoastem%20tipul%20real%20in%20momentul%20compilarii%0A%20%20%20%20creatura.fa_zgomot()%3B%0A%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20creatura%20%3D%20CreaturaMaritima%20%7B%0A%20%20%20%20%20%20%20%20nume%3A%20String%3A%3Afrom(%22Ferris%22)%2C%0A%20%20%20%20%20%20%20%20zgomot%3A%20String%3A%3Afrom(%22blub%22)%2C%0A%20%20%20%20%7D%3B%0A%20%20%20%20generic_fa_zgomot(%26creatura)%3B%0A%7D%0A
+  content_markdown: >
+    Tipurile generice în Rust funcționează mâna în mâna cu trăsăturile. Când descriem un tip parametrizat `T`, putem
+    constrânge ce tipuri pot fi folosite ca argumente,
+    
+    listând trăsăturile necesare (în engleză `trait bound`) pe care argumentul trebuie să le implementeze.
+
+    În acest exemplu, tipul `T` trebuie să implementeze trăsătură `Foo`:
+
+    ```rust
+
+    fn functia_mea<T>(foo: T)
+
+    where
+        T:Foo
+    {
+        ...
+    }
+
+    ```
+
+    Folosind tipurile generice, creăm funcții tipizate static la momentul compilării, ce vor avea tipuri și mărimi
+    cunoscute, permițându-ne să utilizăm expedierea statică și să stocăm sub forma unei valori cu dimensiune.
```

```suggestion
    Folosind tipurile generice, creăm șabloane de funcții care au tipuri de date ce vor fi înlocuite la momentul compilării cu tipuri de date
    cunoscute, permițându-ne să utilizăm apelarea statică.
```

This needs to be rephrased.